runtime/doc/api.txt

```diff
@@ -3360,11 +3360,24 @@ nvim_win_text_height({window}, {opts})                *nvim_win_text_height()*
                   • end_vcol: Ending virtual column index on "end_row",
                     0-based exclusive, rounded up to full screen lines. When
                     omitted include the whole line.
+                  • max_height: Don't add the height of lines below the row
+                    for which this height is reached. Useful to e.g. limit the
+                    height to the window height, avoiding unnecessary work. Or
+                    to find out how many buffer lines beyond "start_row" take
+                    up a certain number of logical lines (returned in
+                    "end_row" and "end_vcol").
 
     Return: ~
         Dict containing text height information, with these keys:
         • all: The total number of screen lines occupied by the range.
         • fill: The number of diff filler or virtual lines among them.
+        • end_row: The row for which "max_height" is reached (first row of a
+          closed fold), or "end_row" from {opts} if "max_height" isn't
+          reached.
+        • end_vcol: Number of columns used up by "end_row" to reach
+          "max_height", or if "max_height" isn't reached, either "end_vcol"
+          from {opts} or the number of columns in "end_row". 0 if "end_row" is
+          a closed fold.
```

Comment:
I think there is some inconsistency with setting `end_vcol` to the number of screen columns. If `end_row` is an empty line, `end_vcol` will always be `0`, but when calling the function `end_vcol = 0` means only including filler lines above `end_row`. A similar problem applies to the case when `end_row` is a folded line, where `end_vcol` is always set to 0.

Of course, there is another problem that it is not possible to tell whether `max_height` is reached at filler lines above `end_row` or at text in `end_row`. Not sure if this should be addressed.

Reply:
Not sure this matters? It still returns the value as documented right? I'll push a doc update to document that supplying 0 end_vcol only includes virtual lines in the first place.

Reply:
Yes it's properly still OK.